src/container/BooksForm.js

```diff
@@ -1,30 +1,57 @@
-import React from 'react';
-
-const BooksForm = () => {
-  const categories = [
-    'Action',
-    'Biography',
-    'History',
-    'Horror',
-    'Kids',
-    'Learning',
-    'Sci-Fi',
-  ];
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { createBook } from '../actions';
+
+const BooksForm = (props) => {
+  const categories = ['Action', 'Biography', 'History', 'Horror', 'Kids', 'Learning', 'Sci-Fi'];
+
+  const [title, setTitle] = useState('');
+  const [category, setCategory] = useState('');
+
+  const handleChange = (e) => {
+    if (e.target.tagName === 'INPUT') {
+      setTitle(e.target.value);
+    } else if (e.target.tagName === 'SELECT') {
+      const { text } = e.target.options[e.target.selectedIndex];
+      setCategory(text);
+    }
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    const bookId = Math.random().toString();
+    const book = {
+      bookId,
+      title,
+      category,
+    };
+    props.createBook(book);
+    setTitle('');
+    setCategory('');
+  };
 
   return (
     <form>
```

Comment:
- [ ] [REQUIRED]  `handleSubmit` should be attached to the form's `onSubmit` event [screenshot from pathwright](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/milestone3/HandleSubmit.png)